File: poo_java/Calculadora1.java
import java.util.Scanner; //Linha 1

public class Calculadora1 {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in); //Linha 6

        System.out.print("Digite o primeiro número: "); //Linha 8
        double numero1 = scanner.nextDouble();          //Linha 9

        System.out.print("Digite o segundo número: ");  //Linha 11
        double numero2 = scanner.nextDouble();          //Linha 12

        System.out.print("Digite a operação (+, -, *, /): "); //Linha 14
        char operacao = scanner.next().charAt(0);             //Linha 15

        double resultado;  //Linha 17

        switch (operacao) { //Linha 19
            case '+':
                resultado = numero1 + numero2;
                break;
            case '-':
                resultado = numero1 - numero2;
                break;
            case '*':
                resultado = numero1 * numero2;
                break;
            case '/':
                if (numero2 == 0) { //Linha 30
                    System.out.println("Erro: divisão por zero!");
                    return;
                }
                resultado = numero1 / numero2;
                break;
            default:
                System.out.println("Operação inválida!");
                return;
        }

        System.out.println("Resultado: " + resultado); //Linha 41
        scanner.close(); //Linha 42
    }
}

/* Linha 1 - import java.util.Scanner; importa a classe Scanner, que permite ler a entrada do usuário.
   Linha 6 - Scanner scanner = new Scanner(System.in); cria um objeto Scanner para ler a entrada do teclado.
   Linhas 8, 11, 14 - Programa solicita ao usuário que digite dois números e a operação desejada (+, -, *, /).
   Linhas 9,12,15 - scanner.nextDouble() lê os números decimais inseridos.
   Linha 17 - O resultado é armazenado na variável resultado.
   Linha 19 - Um switch é usado para determinar qual operação realizar com base no caractere inserido.
   Linha 30 - Se a operação for divisão (/) e o segundo número for zero, o programa exibe uma mensagem de erro e encerra.
   Linha 41 - O resultado da operação é exibido no console.
   Linha 42 - scanner.close(); fecha o objeto Scanner para liberar recursos.
*/

File: node.js/calculadora1.js
const readline = require('readline'); // Importa o módulo readline para ler a entrada do usuário

const rl = readline.createInterface({ // Cria uma interface readline
  input: process.stdin, // Define a entrada como o teclado (stdin)
  output: process.stdout // Define a saída como o console (stdout)
});

function calculadora() { // Define a função calculadora
  rl.question('Digite o primeiro número: ', (num1) => { // Solicita o primeiro número
    rl.question('Digite o segundo número: ', (num2) => { // Solicita o segundo número
      rl.question('Digite a operação (+, -, *, /): ', (operacao) => { // Solicita a operação
        num1 = parseFloat(num1); // Converte a entrada para número de ponto flutuante
        num2 = parseFloat(num2); // Converte a entrada para número de ponto flutuante
        let resultado; // Declara a variável para armazenar o resultado

        switch (operacao) { // Inicia o bloco switch para realizar a operação
          case '+':
            resultado = num1 + num2; // Realiza a adição
            break;
          case '-':
            resultado = num1 - num2; // Realiza a subtração
            break;
          case '*':
            resultado = num1 * num2; // Realiza a multiplicação
            break;
          case '/':
            if (num2 === 0) { // Verifica se o segundo número é zero
              console.log('Erro: Divisão por zero!'); // Exibe mensagem de erro
              rl.close(); // Fecha a interface readline
              return; // Encerra a função
            }
            resultado = num1 / num2; // Realiza a divisão
            break;
          default:
            console.log('Operação inválida!'); // Exibe mensagem de erro para operação inválida
            rl.close(); // Fecha a interface readline
            return; // Encerra a função
        }

        console.log(`Resultado: ${resultado}`); // Exibe o resultado
        rl.close(); // Fecha a interface readline
      });
    });
  });
}

calculadora(); // Chama a função calculadora para iniciar o programa
